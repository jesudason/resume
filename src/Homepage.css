.bg-overlay {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  background-image: -webkit-gradient(
    linear, left top, left bottom, from(rgba(50,50,50,0.8)),
    to(rgba(80,80,80,0.2)), color-stop(.5,#333333)
  );    
}

.welcome-overlay {
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
  position: absolute;
  top: 0;
  left: 0;
  z-index: 2;
  min-height: 100vh; 
}

.welcome-overlay::after {
    content: "";
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 100vh 0 0 40vh;
    border-color: transparent transparent transparent rgba(0, 0, 0, 0.6);
    position: absolute;
    left: 100%;
}

h2 {
  font-size: 48px;
}

.homepage--wrapper {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0  
}

ul {
  padding-left: 0;
}

li.homepage__button {
  background: transparent;
  color: white;
  margin: 12px 24px 0 0;
  padding: 8px 16px;
  border-radius: 5px;
  outline: none;
  list-style: none;
  display: inline-block;
  border: solid white 2px;

}

li.homepage__button:hover {
  background: rgba(0, 0, 0, 0.6);
  color: black;
  -moz-box-shadow: 1px 2px 0px rgba(0, 0, 0, 0.4);
  -webkit-box-shadow: 1px 2px 0px rgba(0, 0, 0, 0.4);
   box-shadow: 1px 2px 0px rgba(0, 0, 0, 0.4);
}

.homepage--text {
  color: #fff;
  position: relative;
  z-index: 5;
  max-width: 540px;
  padding: 0 48px;
}

h2 {
  margin: 0;
}

h4 {
  margin-top: 8px;
  /*font-size: 24px;*/
  margin-bottom: 22px;
}

p {
  font-size: 20px;
  line-height: 1.6;
  margin-bottom: 22px;
}

@media (min-width: 64rem) {

  .homepage--wrapper {
      justify-content: flex-start;
  }

  .welcome-overlay {
    width: 45%;
  }
}
