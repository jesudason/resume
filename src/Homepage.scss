@import "./styles/buttons.scss";
@import "./styles/globals.scss";

// .homepage {
//   background-size: cover !important;
//   overflow: hidden;

//   &__overlay {
//     width: 100%;
//     height: 100%;
//     position: absolute;
//     top: 0;
//     left: 0;
//     background: $background-transparent--light;

    // &--slant {
    //   height: 100%;
    //   width: 100%;
    //   background: $background-transparent;
    //   position: absolute;
    //   top: 0;
    //   left: 0;
    //   z-index: 2;
    //   min-height: 100vh;

    //   &::after {
    //     content: "";
    //     width: 0;
    //     height: 0;
    //     border-style: solid;
    //     border-width: 100vh 0 0 40vh;
    //     border-color: transparent transparent transparent
    //       $background-transparent;
    //     position: absolute;
    //     left: 100%;
    //   }
    // }
  // }

  // &__wrapper {
  //   padding: 21px 25px;
  //   height: 100%;
  //   display: flex;
  //   justify-content: center;

  //   &--text {
  //     z-index: 5;
  //   }
  // }

  // li.homepage__button {
  //   margin: 1em 1em 0 0;
  //   display: inline-block;

  //   &:last-child {
  //     margin-right: 0;
  //   }
  // }
}

// @media (min-width: 48rem) {
//   .homepage {
//     &__wrapper {
//       justify-content: flex-start;
//       // max-width: 21em;
//     }

//     &__overlay {
//       // &--slant {
//       //   width: 45%;
//       // }
//     }
//   }
// }

@media (min-width: 64rem) {
  .homepage {
    &__wrapper {
      display: block;
      // padding: 0;
    }
  }
}
