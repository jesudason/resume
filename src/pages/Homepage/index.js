import React from "react";

import "./style.scss";

class Homepage extends React.Component {
  render() {
    return (
      <div className="homepage">
        <div className="homepage__wrapper">
          <h1>Nicole Jesudason</h1>
          <h2>WEB DEVELOPER</h2>
          <p>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean
            commodo ligula eget dolor. Aenean massa.
          </p>
        </div>
      </div>
    );
  }
}

export default Homepage;
