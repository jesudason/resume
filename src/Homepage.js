import React from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import profilepic from './assets/profile-pic.jpg';
import './Homepage.scss';

class Homepage extends React.Component {
  render() {
    return (  
      <div className="homepage__wrapper">
        <div className="row">
          <div className="col-xs-12 col-sm-3">
            <nav>
              <ul>
                <li className="nav__link">
                  <Link to="/projects/">Projects</Link>
                </li>
                <li className="nav__link">
                  <Link to="/resume/">Resume</Link>
                </li>
                <li className="nav__link">
                  <Link to="/about">About Me</Link>
                </li>
              </ul>
            </nav>
          </div>
          <div className="col-xs-12 col-sm-9">
            <h1>Nicole Jesudason</h1>
            <h2>WEB DEVELOPER</h2>
            <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.</p>
          </div>          
        </div>
      </div>
    );
  }
}

export default Homepage