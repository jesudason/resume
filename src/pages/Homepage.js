import React from "react";

import profilepic from "../assets/profile-pic.jpg";
import Sidenav from "../modules/Sidenav";

import "../Homepage.scss";

class Homepage extends React.Component {
  render() {
    return (
      <div className="homepage__wrapper">
        <div className="row">
          <div className="col-xs-12 col-sm-3">
            <Sidenav />
          </div>
          <div className="col-xs-12 col-sm-9">
            <h1>Nicole Jesudason</h1>
            <h2>WEB DEVELOPER</h2>
            <p>
              Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean
              commodo ligula eget dolor. Aenean massa.
            </p>
          </div>
        </div>
      </div>
    );
  }
}

export default Homepage;
