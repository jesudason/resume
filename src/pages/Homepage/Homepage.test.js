import React from "react";
import { render, cleanup } from "@testing-library/react";
import Homepage from "./";

afterEach(cleanup);

it("should take a snapshot", () => {
  const { asFragment } = render(<Homepage />);

  expect(asFragment(<Homepage />)).toMatchSnapshot();
  //  })
});
