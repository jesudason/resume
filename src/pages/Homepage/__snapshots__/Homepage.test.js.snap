// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <div
      class="homepage__wrapper"
    >
      <h1>
        Nicole Jesudason
      </h1>
      <h2>
        WEB DEVELOPER
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
